File: frontend/src/domain/entities/interview.ts
export interface Interview {
  id: string;
  candidate_id: string;
  interviewer_id: string;
  interview_date: string;
  overall_comment_external: string | null;
  overall_comment_internal: string | null;
  will_text_external: string | null;
  will_text_internal: string | null;
  attract_text_external: string | null;
  attract_text_internal: string | null;
  transcript_raw_text: string | null;
  transcript_source: string | null;
  transcript_url: string | null;
  client_report_markdown: string | null;
  agent_report_markdown: string | null;
  created_at: string;
  updated_at: string;
}

export interface InterviewDetail {
  id: string;
  interview_id: string;
  criteria_item_id: string;
  score_value: number;
  comment_external: string | null;
  comment_internal: string | null;
  created_at: string;
  updated_at: string;
}

export interface InterviewQuestionResponse {
  id: string;
  interview_id: string;
  criteria_item_id: string | null;
  question_text: string;
  answer_summary: string | null;
  hypothesis_text: string | null;
  transcript_reference: string | null;
  is_highlight: boolean;
  created_at: string;
  updated_at: string;
}

export interface InterviewWithDetails extends Interview {
  details: InterviewDetail[];
  question_responses: InterviewQuestionResponse[];
}

export interface InterviewCreate {
  candidate_id: string;
  interviewer_id: string;
  interview_date: string;
}

export interface InterviewUpdate {
  interview_date?: string;
  overall_comment_external?: string;
  overall_comment_internal?: string;
  will_text_external?: string;
  will_text_internal?: string;
  attract_text_external?: string;
  attract_text_internal?: string;
  transcript_raw_text?: string;
  transcript_source?: string;
  transcript_url?: string;
}

export interface InterviewDetailCreate {
  criteria_item_id: string;
  score_value: number;
  comment_external?: string;
  comment_internal?: string;
}

export interface InterviewQuestionResponseCreate {
  criteria_item_id?: string;
  question_text: string;
  answer_summary?: string;
  hypothesis_text?: string;
  transcript_reference?: string;
  is_highlight?: boolean;
}

export const SCORE_LABELS: Record<number, string> = {
  1: '×',
  2: '△',
  3: '◯',
  4: '◎',
};

export const SCORE_OPTIONS = [
  { value: 4, label: '◎' },
  { value: 3, label: '◯' },
  { value: 2, label: '△' },
  { value: 1, label: '×' },
];

File: backend/app/domain/entities/interview.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel


class Interview(BaseModel):
    id: UUID
    candidate_id: UUID
    interviewer_id: UUID
    interview_date: date

    overall_comment_external: str | None = None
    overall_comment_internal: str | None = None

    will_text_external: str | None = None
    will_text_internal: str | None = None
    attract_text_external: str | None = None
    attract_text_internal: str | None = None

    transcript_raw_text: str | None = None
    transcript_source: str | None = None
    transcript_url: str | None = None

    client_report_markdown: str | None = None
    agent_report_markdown: str | None = None

    created_at: datetime
    updated_at: datetime


class InterviewDetail(BaseModel):
    id: UUID
    interview_id: UUID
    criteria_item_id: UUID
    score_value: int  # 1-4 (× = 1, △ = 2, ◯ = 3, ◎ = 4)
    comment_external: str | None = None
    comment_internal: str | None = None
    created_at: datetime
    updated_at: datetime


class InterviewQuestionResponse(BaseModel):
    id: UUID
    interview_id: UUID
    criteria_item_id: UUID | None = None
    question_text: str
    answer_summary: str | None = None
    hypothesis_text: str | None = None
    transcript_reference: str | None = None
    is_highlight: bool = False
    created_at: datetime
    updated_at: datetime

